pxtcompiler/emitter/service.ts

```diff
@@ -1953,6 +1953,15 @@ namespace ts.pxtc.service {
                     }
                 }
 
+                const shadowAttrs = shadowSymbol.attributes;
```

Comment:
Maybe add a quick comment saying "kindNamespace > blockNamespace > namespace" ?

Reply:
block namespace is just for blocks, should not be used for snippets

Reply:
it's for overriding the "real" namespace but in TS that doesn't matter

Reply:
Oh, I thought it would matter in this case as it's being used to look up the kind in `apis.byQName`, so we would want to look it up with the same namespace as a 'key'.

Reply:
How so? The only information on the namespace is toolbox info, right? like color, icon, group names, etc?

Reply:
Just so that it matches and the same default value would be picked for blocks and typescript (as the initial members get picked here: https://github.com/microsoft/pxt/blob/master/pxtlib/service.ts#L522 and the first member found will be the one it defaults to).

I don't think this is an actual case that ever gets hit though, as kinds should have a defined kindNamespace that would take priority; will drop the blockNamespace here